File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {

    public int[] topKFrequent(int[] nums, int k) {
        
        //Count the frquency of each number
        Map<Integer,Integer> map = new HashMap<>();
        for(int n : nums)
        {
            map.put(n,map.getOrDefault(n,0) + 1);
        }

        //Create a min-heap that holds only the k-most frequent element
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b) -> Integer.compare(a[1],b[1]));

        map.forEach((num,freq) -> {

            pq.add(new int[]{num,freq});
            if(pq.size() > k)
                pq.poll();
        });

        int[] ans = new int[k];
        int idx = 0;

        while(!pq.isEmpty())
        {
            ans[idx++] = pq.poll()[0];
        }

        return ans;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count, n = 0,len(nums)
        _map = {}
        total = 0

        for i in range(n):
            total += nums[i]
            if total == k:
                count += 1
            if total - k in _map:
                count += _map[total - k]
            
            _map[total] = _map.get(total,0) + 1
        
        return count

